sklearn/model_selection/_validation.py

```diff
@@ -1167,7 +1182,16 @@ def learning_curve(estimator, X, y, groups=None,
         train_test_proportions = []
         for train, test in cv_iter:
             for n_train_samples in train_sizes_abs:
-                train_test_proportions.append((train[:n_train_samples], test))
+                if stratify and n_train_samples < n_max_training_samples:
+                    inner_cv = StratifiedShuffleSplit(
```

Comment:
why not ``train_test_split``? I kind of expect that to be slightly shorter?

Reply:
Indeed.